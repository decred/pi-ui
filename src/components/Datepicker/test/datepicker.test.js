import React from "react";
import DatePicker from "../Datepicker";
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";

describe("DatePicker component", () => {
  test("Matches the snapshot", () => {
    const datePicker = create(<DatePicker value={{ year: 2020, month: 12 }} />);
    expect(datePicker.toJSON()).toMatchSnapshot();
  });

  test("lang prop values are used as months", () => {
    const years = {
      min: { year: 2018, month: 1, day: 25 },
      max: { year: 2020, month: 2, day: 4 },
    };
    const monthsTranslations = [
      "Jan",
      "Feb",
      "Mar",
      "Apr",
      "May",
      "Jun",
      "Jul",
      "Aug",
      "Sep",
      "Oct",
      "Nov",
      "Dec",
    ];
    const { queryByText, rerender, unmount } = render(
      <DatePicker
        lang={monthsTranslations}
        years={years}
        value={{ year: 2019, month: 11, day: 15 }}
        show={true}>
        <></>
      </DatePicker>
    );
    // Selected value is 15.11.2019 => Nov month is displayed.
    expect(queryByText(/Nov/i)).toBeTruthy();
    unmount();
    rerender(
      <DatePicker
        lang={monthsTranslations}
        years={years}
        value={{ year: 2019, month: 12, day: 15 }}
        show={true}>
        <></>
      </DatePicker>
    );
    // Selected value is 15.12.2019 => Nov month is displayed.
    expect(queryByText(/Dec/i)).toBeTruthy();
  });
});
