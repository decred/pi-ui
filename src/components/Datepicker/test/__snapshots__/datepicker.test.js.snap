// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component Matches the snapshot 1`] = `
<div
  className="monthPicker"
>
  <span
    className="defaultDatepickerValueWrapper"
    onClick={[Function]}
  >
    12/2020
    <div
      className="defaultDatepickerArrow"
    />
  </span>
  <div
    className="rmpContainer rmpTable"
  >
    <div
      className="rmpOverlay"
      onClick={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "KhtmlUserSelect": "none",
          "MozUserSelect": "none",
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "WebkitTouchCallout": "none",
          "WebkitUserSelect": "none",
          "msUserSelect": "none",
          "userSelect": "none",
        }
      }
    />
    <div
      className="rmpCell"
    >
      <div
        className="rmpPopup"
      />
    </div>
  </div>
</div>
`;
